### PR TITLE
Error handling and return report to user

### PR DESCRIPTION
**Pull request by vkbo**
**Summary**:

This PR changes the way the validator handles errors when checking the MMD XML file. Instead of writing issues to the logger object, the errors are added to lists and can be returned to the user via the API.

**Related issue**:

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: dmci/api/worker.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from lxml import etree
 
 from dmci import CONFIG
-from dmci.mmd_tools import full_check
```
Comment (mortenwh) [inline dmci/api/worker.py:27]: can we rename `mmd_tools` to, e.g., `validator` or something?

**Review by mortenwh — CHANGES_REQUESTED** [review]
It looks good but can we rename the `mmd_tools` sub-package? Can you also connect the PR to an issue, or estimate story points?

**Comment by vkbo** [comment]
This PR is still in draft. I've discovered multiple issues with the rectangle checker today that can cause the API to crash, so I'm rewriting it.

File: dmci/api/worker.py (hunk 1 above)
Comment (vkbo) [reply dmci/api/worker.py:27]: I changed it to just `tools`.

File: dmci/api/worker.py [hunk 2]
```diff
@@ -111,13 +111,15 @@ def distribute(self):
                 **self._kwargs
             )
             valid &= obj.is_valid()
-            if obj.is_valid:
+            if obj.is_valid():
```
Comment (vkbo) [inline dmci/api/worker.py:114]: This was a bug

**Review by mortenwh — CHANGES_REQUESTED** [review]
Thanks - it looks very good :) does it close any issues? Or could you add an estimate?

**Comment by vkbo** [comment]
It resolves parts of existing issues. Let me reshuffle the relevant issues a bit, and maybe add a new one as well.

File: dmci/api/app.py [hunk 3]
```diff
@@ -55,33 +55,40 @@ def __init__(self):
...
-            result, msg = worker.validate(data)
-            if result:
-                return self._persist_file(data, full_path)
+            valid, msg = worker.validate(data)
+            if not valid:
+                return msg, 400
+
+            # Run the distributors
+            err = []
+            status, valid, _, failed, skipped = worker.distribute()
```
[the remark this answers is not in the document]
Comment (vkbo) [reply dmci/api/app.py:82]: Yes, it does now.

[the remark this answers is not in the document]
Comment (vkbo) [reply dmci/api/app.py:82]: Well, it doesn't block until success, it blocks until all distributors have run and returned a status code. Whatever that status code may be.

**Comment by vkbo** [comment]
Could you please finish the reviews so we can move forward with this PR?

**Review by mortenwh — APPROVED** [review]

**Pull request merged**